libcaf_core/test/actor_system_config.cpp

```diff
@@ -150,6 +149,21 @@ CAF_TEST(file input overrides defaults but CLI args always win) {
   CHECK_EQ(content(cfg), res);
 }
 
+CAF_TEST(parsing - with config file cli option) {
+  MESSAGE("Try opening non-existent config file");
+  cfg.clear();
+  auto err = cfg.parse(string_list{"--config-file=test-me"});
+  CHECK_EQ(err, caf::sec::cannot_open_file);
+  CHECK(cfg.remainder.empty());
+  CHECK_EQ(get_or(cfg, "config-file", ""), "test-me");
+
+  cfg.clear();
+  err = cfg.parse(string_list{"--config-file", "test-me"});
+  CHECK_EQ(err, caf::sec::cannot_open_file);
+  CHECK(cfg.remainder.empty());
+  CHECK_EQ(get_or(cfg, "config-file", ""), "test-me");
+}
+
```

Comment:
Unit tests shouldn't do file I/O. Did you consider adding this as a test to `robot/config/dump-config.robot`? The program should print something like `error while parsing CLI and file options: caf::sec::cannot_open_file("test-me")`, so we could just check the output contians `cannot_open_file`.

Reply:
I've done robot. I'll remove this.